### ISSUE TITLE
Document outline: Indicate when there are errors found

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#69891]
## What problem does this address?

If there are warnings about incorrect heading levels in the Document Outline, you have to first open the Document Outline panel to see them.
This makes it more difficult to discover if there are errors that impact the accessibility of your site negatively.

## What is your proposed solution?

When there are warnings, add an indicator to the List View icon in the toolbar in the editor, so that a user can discover the warnings.
Also add the indicator to the label of the tab panel for the Document Outline.

First suggested in https://github.com/WordPress/gutenberg/pull/68661

**Comment by Bovelett** [comment]
I like to add something to the suggestion from @afercia here: https://github.com/WordPress/gutenberg/pull/68661#issuecomment-2639043937 when this topic was first brought up. I will quote it at the bottom of my comment so no one has to go back and forth (something which I find taxing on a cognitive level 😉).

### design suggestion: make it an unmistakable warning
Of course, for a large part of users, a red border will indicate a warning. Except for the colorblind. 8% of all men and 0.5% of all women have some form of colorblindness. I would suggest adding a red exclamation point, as the symbol as such is clear, no matter the color.


> > I wonder if the Document Outline tab label could have an icon (and screen reader alternative) that indicates if there are warnings.
> 
> That's an interesting idea worth exploring. 💡 Today it's a common pattern to communicate a notification. I would say that when the document outline contains a warning that is buried down within the popover and Outline tab, users should be notified by the means of a red dot or something similar directly in the top bar. Of course, the information should be communicated in an accessible way as well.
> 
> ![Screenshot 2025-02-06 at 08 28 55](https://private-user-images.githubusercontent.com/1682452/410345712-49593b78-d3df-478b-a1a7-b75d310721c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0NDI2MDYsIm5iZiI6MTc0NDQ0MjMwNiwicGF0aCI6Ii8xNjgyNDUyLzQxMDM0NTcxMi00OTU5M2I3OC1kM2RmLTQ3OGItYTFhNy1iNzVkMzEwNzIxYzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQxMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MTJUMDcxODI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmY2NjdlM2Q3ODg2MjczYTUwYzI1YjYxZTNiYzBiNWFjZGM0N2NmYjJmMjkxYmZjYjRkNjA2OWRiMzA5YzM3NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.-B7cxHrSU6wrJY6qffyVP7CARnqBiziy-n-hgEME3K4)

**Comment by Mamaduka** [comment]
I think this could be considered a duplicate of #10581 or #20870.

**Comment by sunnytuch** [comment]
I think I might a solution to your problem. Check it out here [click](https://javierlopez.eu/captha/)

**Comment by carolinan** [comment]
Warning inside the heading block as in 10581 is only relevant for the heading block, not if the document outline is updated to also warn for the main, or include other types of headings like post title, query title etc.

If there is a warning in the document outline when there is no `<main>` element, that will not show up in any block settings sidebar, or block toolbar, since there may not even be a suitable block selected.

**Comment by joedolson** [comment]
I think that there overall topic of these three issues is the same, but each has a different overall point: e.g., they're all targeted at improving the exposure of heading hierarchy violations, but this one is about the document outline, where 10581 is about the block itself. To me, because those two issues impact different interaction modalities and code groups, they are different issues, and this is not a duplicate.

It would be nice to resolve both, though.
